spacy_curated_transformers/cli/fill_config_transformer.py

```diff
@@ -0,0 +1,469 @@
+                    f"Hugging Face model config has a missing key '{hf_key}'", exits=1
+                )
+        elif source == HfParamSource.TOKENIZER_CONFIG:
+            value = getattr(hf_tokenizer, hf_key, None)
+            if value is None:
+                msg.fail(
+                    f"Hugging Face tokenizer config has a missing key '{hf_key}'",
+                    exits=1,
+                )
+        filled_params[param_name] = value
+
+    msg.info(title="Filled-in model parameters:")
+    msg.table(filled_params)
+
+    return filled_params
+
+
+def _create_intermediate_configs(config: Config, dot_path: str):
+    splits = dot_path.split(".")
+    current = config
+    while splits:
+        name = splits[0]
+        if name not in current:
+            current[name] = {}
+        current = current[name]
+        splits = splits[1:]
+
+
+def _save_encoder_loader_config(
+    msg: Printer,
+    config: Config,
+    transformer_name: str,
+    model_name: str,
+    model_revision: str,
+):
+    _create_intermediate_configs(
+        config, f"initialize.components.{transformer_name}.encoder_loader"
+    )
+    inner = config["initialize"]["components"][transformer_name]["encoder_loader"]
+    if inner:
+        msg.warn(f"Overwriting transformer encoder loader config")
+        inner.clear()
+
+    inner.update(
+        {
+            "@model_loaders": "spacy-curated-transformers.HFTransformerEncoderLoader.v1",
+            "name": model_name,
+            "revision": model_revision,
+        }
+    )
+
+
+def _save_piece_loader_config(
+    msg: Printer,
+    config: Config,
+    transformer_name: str,
+    model_name: str,
+    model_revision: str,
+    *,
+    overwrite: bool,
+):
+    _create_intermediate_configs(
+        config, f"initialize.components.{transformer_name}.piece_loader"
+    )
+    inner = config["initialize"]["components"][transformer_name]["piece_loader"]
+    if inner:
+        if overwrite:
+            msg.warn(f"Overwriting piece encoder loader config")
+            inner.clear()
+        else:
+            arch = inner.get("@model_loaders")
+            if (
+                arch == "spacy-curated-transformers.HFPieceEncoderLoader.v1"
+                and inner["name"] == model_name
+                and inner.get("revision", "main") == model_revision
+            ):
+                return
+
+            msg.warn(
+                f"Existing piece encoder loader might not be compatible with model '{model_name}'"
+            )
+            return
+
+    inner.update(
+        {
+            "@model_loaders": "spacy-curated-transformers.HFPieceEncoderLoader.v1",
+            "name": model_name,
+            "revision": model_revision,
+        }
+    )
+
+
+def _get_hf_model_config(name: str, revision: str) -> Dict[str, Any]:
+    from huggingface_hub import hf_hub_download
+
+    config_path = hf_hub_download(
+        repo_id=name, filename="config.json", revision=revision
+    )
```

Comment:
Try running this with something like `--model-name foobar` and you'd get a long and ugly error stack trace. Can we catch that, provide a commandline-friendly message and explain to the user what this value should look like?

Reply:
Added a `try...except` block to handle errors. The error message has to be somewhat under-specified as a number of things could potentially fail during this process (on HF Hub's end) is large.